pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -164,3 +165,46 @@ func validateMachine(machine machinev1.Machine, providerSpec v1beta1.GCPMachineP
 	// Default values can also be set here
 	return nil
 }
+
+// Returns true if machine exists.
+func (r *Reconciler) instanceExists() (bool, error) {
+	if err := validateMachine(*r.machine, *r.providerSpec); err != nil {
+		return false, fmt.Errorf("failed validating machine provider spec: %v", err)
+	}
+	zone := r.providerSpec.Zone
+	// Need to verify that our project/zone exists before checking machine, as
+	// invalid project/zone produces same 404 error as no machine.
+	err := r.validateZone()
+	if err != nil {
+		klog.Errorf("Unable to verify project/zone exists: %v/%v", r.projectID, zone)
+		return false, err
+	}
+
+	klog.Errorf("%v, %v, %v", r.projectID, zone, r.machine.Name)
+	_, err = r.computeService.InstancesGet(r.projectID, zone, r.machine.Name)
+	if err != nil {
+		notFound := isNotFoundError(err)
```

Comment:
I think this may be clearer expressed as:

```go
	klog.Errorf("%v, %v, %v", r.projectID, zone, r.machine.Name)
	if _, err = r.computeService.InstancesGet(r.projectID, zone, r.machine.Name); err == nil {
		return true, nil
	}
	if isNotFoundError(err) {
		klog.Errorf("%v", err)
		return false, nil
	}
	klog.Errorf("Error getting running instances: %v", err)
	return false, err
```